tools/cassandra/cqlclient.go

```diff
@@ -111,26 +113,36 @@ const (
 
 var _ schema.SchemaClient = (*CqlClientImpl)(nil)
 
-// NewCQLClient returns a new instance of CQLClient
 func NewCQLClient(cfg *CQLClientConfig, expectedConsistency gocql.Consistency) (CqlClient, error) {
-	var err error
+	retrier := createClientCreationRetrier()
+	cassClient := gocql.GetRegisteredClient()
+	return NewCQLClientWithRetry(cfg, expectedConsistency, cassClient, retrier)
+}
 
+// NewCQLClient returns a new instance of CQLClient
+func NewCQLClientWithRetry(cfg *CQLClientConfig, expectedConsistency gocql.Consistency, cassClient gocql.Client, retrier *backoff.ThrottleRetry) (CqlClient, error) {
 	cqlClient := new(CqlClientImpl)
 	cqlClient.cfg = cfg
 	cqlClient.nReplicas = cfg.NumReplicas
-	cqlClient.session, err = gocql.GetRegisteredClient().CreateSession(gocql.ClusterConfig{
-		Hosts:                 cfg.Hosts,
-		Port:                  cfg.Port,
-		User:                  cfg.User,
-		Password:              cfg.Password,
-		AllowedAuthenticators: cfg.AllowedAuthenticators,
-		Keyspace:              cfg.Keyspace,
-		TLS:                   cfg.TLS,
-		Timeout:               time.Duration(cfg.Timeout) * time.Second,
-		ConnectTimeout:        time.Duration(cfg.ConnectTimeout) * time.Second,
-		ProtoVersion:          cfg.ProtoVersion,
-		Consistency:           expectedConsistency,
-	})
+
+	op := func() error {
+		var err error
+		cqlClient.session, err = cassClient.CreateSession(gocql.ClusterConfig{
+			Hosts:                 cfg.Hosts,
+			Port:                  cfg.Port,
+			User:                  cfg.User,
+			Password:              cfg.Password,
+			AllowedAuthenticators: cfg.AllowedAuthenticators,
+			Keyspace:              cfg.Keyspace,
+			TLS:                   cfg.TLS,
+			Timeout:               time.Duration(cfg.Timeout) * time.Second,
+			ConnectTimeout:        time.Duration(cfg.ConnectTimeout) * time.Second,
+			ProtoVersion:          cfg.ProtoVersion,
+			Consistency:           expectedConsistency,
+		})
+		return err
+	}
+	err := retrier.Do(context.Background(), op)
```

Comment:
nit: don't use a variable, just put the function here. I'm unsure if the Go compiler is smart enough, but it should not be the problem.

Reply:
Removed variable